shared/trieutil/sparse_merkle.go

```diff
@@ -183,15 +185,14 @@ func (m *SparseMerkleTrie) ToProto() *protodb.SparseMerkleTrie {
 // VerifyMerkleBranch verifies a Merkle branch against a root of a trie.
 func VerifyMerkleBranch(root []byte, item []byte, merkleIndex int, proof [][]byte) bool {
 	node := bytesutil.ToBytes32(item)
-	currentIndex := merkleIndex
-	for i := 0; i < len(proof); i++ {
-		if currentIndex%2 != 0 {
+	for i := 0; i < int(params.BeaconConfig().DepositContractTreeDepth); i++ {
```

Comment:
`params.BeaconConfig().DepositContractTreeDepth` should be a parameter

See `depth: uint64` in the spec definition